src/backend/joanie/core/factories.py

```diff
@@ -688,6 +688,22 @@ class Meta:
     owner = factory.SubFactory(UserFactory)
 
 
+class OrganizationAddressFactory(factory.django.DjangoModelFactory):
+    """A factory to create an organization address"""
+
+    class Meta:
+        model = models.Address
+
+    title = factory.fuzzy.FuzzyChoice(["Organization", "University"])
+    address = factory.Faker("street_address")
+    postcode = factory.Faker("postcode")
+    city = factory.Faker("city")
+    country = factory.Faker("country_code")
+    first_name = factory.Faker("first_name")
+    last_name = factory.Faker("last_name")
+    organization = factory.SubFactory(OrganizationFactory)
```

Comment:
Could be simplified to 
```suggestion
class OrganizationAddressFactory(UserAddressFactory):
    """A factory to create an organization address"""

    title = factory.fuzzy.FuzzyChoice(["Organization", "University"])
    owner = None
    organization = factory.SubFactory(OrganizationFactory)
```
FMI, why the Address model has first_name and last_name attributes, as it is already in the User model, and probably somewhere in the Organization one ?

Reply:
It's for the name that could be present on the letter box. It might differ if it is not your own address. The thought was if I had a delivery address at mines, the first name and last name would be mine, but if I'm used to deliver it at my parents house, it would be my parent's name as first name and last name.

Reply:
We made finally made a base class for the Address factory, and there are 2 subclasses one for `UserAddressFactory`, and another for `OrganizationAddressFactory`.

```python
class AddressFactory(factory.django.DjangoModelFactory):
    """A factory to create an address"""

    class Meta:
        model = models.Address

    title = factory.fuzzy.FuzzyChoice(["Home", "Office"])
    address = factory.Faker("street_address")
    postcode = factory.Faker("postcode")
    city = factory.Faker("city")
    country = factory.Faker("country_code")
    first_name = factory.Faker("first_name")
    last_name = factory.Faker("last_name")


class UserAddressFactory(AddressFactory):
    """A factory to create a user address"""

    owner = factory.SubFactory(UserFactory)


class OrganizationAddressFactory(AddressFactory):
    """A factory to create an organization address"""

    organization = factory.SubFactory(OrganizationFactory)

```